From: Edward Cree <ecree@solarflare.com>
Date: Wed, 25 Sep 2013 17:34:12 +0100
Subject: [PATCH v2 148/244] sfc: Fix internal indices of ethtool stats for
 EF10
Patch-mainline: v3.12-rc7
Git-commit: 87648cc925222f16c3bd119af8ae8e8fb2f3fe1d
References: bsc#909618 FATE#317521

The indices in nic_data->stats need to match the EF10_STAT_whatever
enum values.  In efx_nic_update_stats, only mask; gaps are removed in
efx_ef10_update_stats.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/sfc/nic.c |    9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

--- a/drivers/net/ethernet/sfc/nic.c
+++ b/drivers/net/ethernet/sfc/nic.c
@@ -479,8 +479,7 @@ size_t efx_nic_describe_stats(const stru
  * @count: Length of the @desc array
  * @mask: Bitmask of which elements of @desc are enabled
  * @stats: Buffer to update with the converted statistics.  The length
- *	of this array must be at least the number of set bits in the
- *	first @count bits of @mask.
+ *	of this array must be at least @count.
  * @dma_buf: DMA buffer containing hardware statistics
  * @accumulate: If set, the converted values will be added rather than
  *	directly stored to the corresponding elements of @stats
@@ -513,11 +512,9 @@ void efx_nic_update_stats(const struct e
 			}
 
 			if (accumulate)
-				*stats += val;
+				stats[index] += val;
 			else
-				*stats = val;
+				stats[index] = val;
 		}
-
-		++stats;
 	}
 }
